src/ess/polarization/correction.py

```diff
@@ -0,0 +1,168 @@
+    """
+    Up-up column of combined correction coefficients for polarizer and analyzer.
+
+    This is effectively a column resulting from a sparse matrix-matrix product.
+
+    This function returns the up-up column.
+
+    Parameters
+    ----------
+    analyzer :
+        Correction coefficients for the analyzer.
+    polarizer :
+        Correction coefficients for the polarizer.
+
+    Returns
+    -------
+    :
+        Combined correction coefficients.
+    """
+    return PolarizationCorrection[Up, Up](
+        upup=polarizer.diag * analyzer.diag,
+        updown=polarizer.diag * analyzer.off_diag,
+        downup=polarizer.off_diag * analyzer.diag,
+        downdown=polarizer.off_diag * analyzer.off_diag,
+    )
+
+
+def compute_polarization_correction_updown(
+    analyzer: PolarizingElementCorrection[Up, Down, Analyzer],
+    polarizer: PolarizingElementCorrection[Up, Down, Polarizer],
+) -> PolarizationCorrection[Up, Down]:
+    """
+    Up-down column of combined correction coefficients for polarizer and analyzer.
+
+    See :py:func:`compute_polarization_correction_upup` for more details.
+    """
+    return PolarizationCorrection[Up, Down](
+        upup=polarizer.diag * analyzer.off_diag,
+        updown=polarizer.diag * analyzer.diag,
+        downup=polarizer.off_diag * analyzer.off_diag,
+        downdown=polarizer.off_diag * analyzer.diag,
+    )
+
+
+def compute_polarization_correction_downup(
+    analyzer: PolarizingElementCorrection[Down, Up, Analyzer],
+    polarizer: PolarizingElementCorrection[Down, Up, Polarizer],
+) -> PolarizationCorrection[Down, Up]:
+    """
+    Down-up column of combined correction coefficients for polarizer and analyzer.
+
+    See :py:func:`compute_polarization_correction_upup` for more details.
+    """
+    return PolarizationCorrection[Down, Up](
+        upup=polarizer.off_diag * analyzer.diag,
+        updown=polarizer.off_diag * analyzer.off_diag,
+        downup=polarizer.diag * analyzer.diag,
+        downdown=polarizer.diag * analyzer.off_diag,
+    )
+
+
+def compute_polarization_correction_downdown(
+    analyzer: PolarizingElementCorrection[Down, Down, Analyzer],
+    polarizer: PolarizingElementCorrection[Down, Down, Polarizer],
+) -> PolarizationCorrection[Down, Down]:
+    """
+    Down-down column of combined correction coefficients for polarizer and analyzer.
+
+    See :py:func:`compute_polarization_correction_upup` for more details.
+    """
+    return PolarizationCorrection[Down, Down](
+        upup=polarizer.off_diag * analyzer.off_diag,
+        updown=polarizer.off_diag * analyzer.diag,
+        downup=polarizer.diag * analyzer.off_diag,
+        downdown=polarizer.diag * analyzer.diag,
+    )
+
+
+def compute_polarization_corrected_data(
+    channel: ReducedSampleDataBySpinChannel[PolarizerSpin, AnalyzerSpin],
+    polarization_correction: PolarizationCorrection[PolarizerSpin, AnalyzerSpin],
+) -> PolarizationCorrectedData[PolarizerSpin, AnalyzerSpin]:
```

Comment:
Don't we want to sum the contributions from all `PolarizerSpin, AnalyzerSpin` combinations? Or is that done in a later step?

Reply:
As you mentioned in the PR description, doing it this way and combining in a later step will increase memory consumption, it might also be less complicated to just do the product here instead of splitting it in two steps.

Reply:
Than can be done later. Functionality for that will be added in future work.